kie-maven-plugin/src/main/java/org/kie/maven/plugin/ValidateDMNMojo.java

```
@@ -0,0 +1,176 @@
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.drools.compiler.kproject.models.KieModuleModelImpl;
+import org.kie.api.builder.Message.Level;
+import org.kie.api.builder.model.KieModuleModel;
+import org.kie.dmn.api.core.DMNMessage;
+import org.kie.dmn.core.assembler.DMNAssemblerService;
+import org.kie.dmn.core.compiler.DMNProfile;
+import org.kie.dmn.feel.util.ClassLoaderUtil;
+import org.kie.dmn.model.api.DMNModelInstrumentedBase;
+import org.kie.dmn.model.api.Definitions;
+import org.kie.dmn.validation.DMNValidator;
+import org.kie.dmn.validation.DMNValidator.Validation;
+import org.kie.dmn.validation.DMNValidatorFactory;
+import org.kie.internal.utils.ChainedProperties;
+
+@Mojo(name = "validateDMN",
+      requiresDependencyResolution = ResolutionScope.NONE,
+      requiresProject = true,
+      defaultPhase = LifecyclePhase.PROCESS_RESOURCES)
+public class ValidateDMNMojo extends AbstractKieMojo {
+
+    @Parameter(required = true, defaultValue = "${project.build.resources}")
+    private List<Resource> resources;
+
+    @Parameter
+    private Map<String, String> properties;
+
+    @Parameter(required = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    @Parameter(property = "validateDMN", defaultValue = "VALIDATE_SCHEMA,VALIDATE_MODEL")
+    private String validateDMN;
+    private List<Validation> actualFlags = new ArrayList<>();
+
+    private DMNValidator validator;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        try {
+            for (String p : validateDMN.split(",")) {
+                actualFlags.add(Validation.valueOf(p));
+            }
+        } catch (IllegalArgumentException e) {
+            // do nothing.
```

This seems suspicious. Maybe it would be better to rethrow this exception or at least log something.

---

The idea was to allow flags which are not validator flag such as `disabled`

---

Question because I'm not much familiar with these flags. Isn't there any other possible method which is able to recognize non validation flag other than catching exception? If there isn't maybe we can add more descriptive comment than "do nothing". For example "Allow non validator flags to pass".